tests/System.Text.Primitives.Tests/EncodingPerfComparisonTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text.Utf8;
+using Xunit;
+using Microsoft.Xunit.Performance;
+
+namespace System.Text.Primitives.Tests
+{
+    public class EncodingPerfComparisonTests
+    {
+        [Benchmark]
+        [InlineData(false, 1000, 0x0020, 0x007F)]
+        [InlineData(false, 1000, 0x0080, 0x07FF)]
+        [InlineData(false, 1000, 0x0800, 0xd7FF)]
+        [InlineData(false, 1000, 0x0020, 0xd7FF)]
+        [InlineData(true, 1000, 0x0020, 0x007F)]
+        [InlineData(true, 1000, 0x0080, 0x07FF)]
+        [InlineData(true, 1000, 0x0800, 0xd7FF)]
+        [InlineData(true, 1000, 0x0020, 0xd7FF)]
+        public void EncodingPerformanceTestUsingCoreCLR(bool convertToArrayInLoop, int charLength, int minCodePoint, int maxCodePoint)
+        {
+            string unicodeString = GenerateString(charLength, minCodePoint, maxCodePoint);
+            Console.WriteLine(unicodeString);
+            ReadOnlySpan<char> characters = unicodeString.Slice();
+
+            Encoding utf8 = Encoding.UTF8;
+
+            int utf8Length = utf8.GetByteCount(unicodeString);
+            byte[] utf8Buffer = new byte[utf8Length];
+            Span<byte> span;
+
+            char[] charArray = characters.ToArray();
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                    if (convertToArrayInLoop)
+                        utf8.GetBytes(characters.ToArray(), 0, characters.Length, utf8Buffer, 0);
```

Comment:
> the performance if you convert to array in every loop iteration.

It seems, there is contention here due to difference of opinion. I am fine either way (keeping it or removing it) when an agreement is reached.

One thing I would like clarification, you stated that repeated conversion within a loop wouldn't happen. However, isn't this equivalent to a single conversion in user code? The loop is there to take multiple samples, no?